modules/test/playwright/tests/frontend-data-set-web/main/advanced.spec.ts

```diff
@@ -1161,6 +1161,169 @@ test(
+			const firstTableRow = fdsSamplePage.table.bodyRows.first();
+
+			await firstTableRow.hover();
+
+			await expect(firstRowSampleEditQuickActionLink).toBeVisible();
+		});
+
+		await test.step('When clicking on the ellipsis and hovering over another row, multiple quick action menus are displayed', async () => {
+			await thirdRowItemActionButton.hover();
+
+			await expect(thirdRowSampleEditQuickActionLink).toBeVisible();
+
+			await thirdRowItemActionButton.hover();
+
+			await thirdRowItemActionButton.click();
+
+			await expect(page.locator('.dropdown-menu.show')).toBeVisible();
+
+			await fdsSamplePage.table.bodyRows.first().hover();
+
+			await expect(
+				page
+					.locator('.fds table tbody tr')
+					.first()
+					.getByLabel('Sample Edit')
+			).toBeVisible(); // `page` must be used here since using `fdsSamplePage` won't locate the element with the dropdown open
+
+			await firstTableHeadCell.click(); // Close the dropdown
+		});
+
+		await test.step('Assert quick action can be displayed on only one active row', async () => {
+			await firstTableHeadCell.hover();
+
+			await firstTableHeadCell.click();
+
+			await firstRowItemActionButton.hover();
+
+			await expect(firstRowSampleEditQuickActionLink).toBeVisible();
+
+			await thirdRowItemActionButton.hover();
+
+			await thirdRowItemActionButton.click();
+
+			await expect(firstRowSampleEditQuickActionLink).not.toBeVisible();
+
+			await firstTableHeadCell.click(); // Close the dropdown
+		});
+
+		await test.step('Assert that quick action icons list should be limited to three actions', async () => {
+			await firstRowItemActionButton.click();
+
+			const sampleViewButton = fdsSamplePage.table.bodyRows
+				.getByRole('cell', {name: 'Actions'})
+				.getByLabel('Sample View');
+
+			const sampleEditButton = fdsSamplePage.table.bodyRows
+				.getByRole('cell', {name: 'Actions'})
+				.getByLabel('Sample Edit');
+
+			const sampleDelete = fdsSamplePage.table.bodyRows
+				.getByRole('cell', {name: 'Actions'})
+				.getByLabel('Sample Delete');
+
+			await expect(sampleViewButton).toBeVisible();
+
+			await expect(sampleEditButton).toBeVisible();
+
+			await expect(sampleDelete).toBeVisible();
+		});
+
+		await test.step('Assert that test-pencil is appended to browser URL after clicking', async () => {
+			await firstTableHeadCell.hover();
+
+			await firstTableHeadCell.click();
+
+			await firstRowItemActionButton.hover();
+
+			await firstRowSampleEditQuickActionLink.click();
+
+			expect(page.url()).toContain('#test-pencil');
+		});
+
+		await test.step('Assert the quick action is not visible when the row checkbox is checked', async () => {
+			await fdsSamplePage.table.bodyRows
+				.first()
+				.getByRole('checkbox')
+				.click();
+
+			await firstRowItemActionButton.hover();
+
+			await expect(firstRowSampleEditQuickActionLink).not.toBeVisible();
+		});
```

Comment:
This part will cause the test to fail because quick actions are still visible when the checkbox is checked. Created bug ticket: https://liferay.atlassian.net/browse/LPD-54629